File: Challenges/Javascript/happyNumber.js
function happyNumber(n) {
    let set = [];
    while(!set.includes(n)) {
        let sum = 0;
        set.push(n);
        
        for(let e of n+"") sum += parseInt(e) ** 2;
        
        if(sum == 1) return true;
        else n = sum;
    }
    
    return false;
}

File: Challenges/Java/HappyNumber.java
import java.util.Set;
import java.util.HashSet;

final class HappyNumber {

	boolean happyNumber(int n) {
	    Set<Integer> set = new HashSet<>();
	    
	    while(set.add(n)) {
	        int v = (n+"").chars().map(e -> new Integer(((char) e)+""))
	        .map(e -> e * e).sum();
	        
	        if(v == 1) return true;
	        else n = v;
	        
	    }
	    return false;
	}
}